models/base_model.py

```diff
@@ -0,0 +1,160 @@
+import os
+import torch
+from collections import OrderedDict
+from . import networks
+
+
+class BaseModel:
+
+    # modify parser to add command line options,
+    # and also change the default values if needed
+    @staticmethod
+    def modify_commandline_options(parser, is_train):
+        return parser
+
+    def name(self):
+        return "BaseModel"
+
+    def initialize(self, opt):
+        self.opt = opt
+        self.use_gpu = opt.model.use_gpu
+        self.device = torch.device(
+            "cuda" if (torch.cuda.is_available() and opt.model.use_gpu) else "cpu"
+        )
+        self.isTrain = opt.model.is_train
+        self.loss_names = []
+        self.model_names = []
+        self.comet_exp = opt.comet_exp
+        self.save_dir = opt.train.output_dir + "/checkpoints"
+
+    def set_input(self, input):
+        pass
+
+    def forward(self):
+        pass
+
+    def validate(self):
+        pass
+
+    # load and print networks; create schedulers
+    def setup(self):
+        # TODO Resume training
+
+        if not self.isTrain or self.opt.train.resume_checkpoint:
+            load_suffix = "iter_%d" % self.opt.train.load_iter
+            self.load_networks(load_suffix)
+
+        self.print_networks(self.opt.model.verbose)
+
+    # make models eval mode during test time
+    def eval(self):
+        for name in self.model_names:
+            if isinstance(name, str):
+                net = getattr(self, "net" + name)
+                net.eval()
+
+    # used in test time, wrapping `forward` in no_grad() so we don't save
+    # intermediate steps for backprop
+    def test(self):
+        with torch.no_grad():
+            self.forward()
+
+    def optimize_parameters(self):
+        pass
+
+    # update learning rate (called once every epoch)
+    def update_learning_rate(self):
```

Comment:
This is syntactically correct but you're lucky the docstring counts ; this should `raise` something or at least `pass`